hibernate-core/src/main/java/org/hibernate/graph/internal/RootGraphImpl.java

```diff
@@ -43,22 +55,51 @@ public String getName() {
 
 	@Override
 	public RootGraphImplementor<J> makeCopy(boolean mutable) {
-		return new RootGraphImpl<>( null, this, mutable);
+		return new RootGraphImpl<>( null, this, mutable );
 	}
 
 	@Override
 	public SubGraphImplementor<J> makeSubGraph(boolean mutable) {
-		return new SubGraphImpl<>(this, mutable);
+		return new SubGraphImpl<>( this, mutable );
 	}
 
 	@Override
 	public RootGraphImplementor<J> makeRootGraph(String name, boolean mutable) {
 		return !mutable && !isMutable() ? this : super.makeRootGraph( name, mutable );
 	}
 
+	public <S extends J> SubGraphImplementor<S> addTreatedSubgraph(Class<S> type) {
+		ManagedDomainType<S> subTypeEntityDomainType = DomainModelHelper.findSubType( super.managedType, type );
+
+		SubGraphImplementor<S> subgraph = new SubGraphImpl<>( subTypeEntityDomainType, true );
+
+		subclassSubgraphs.putIfAbsent( subTypeEntityDomainType, subgraph );
+
+		return subgraph;
```

Comment:
The sub-graph needs to know about `this` since the javadoc of `addTreatedSubgraph` says:

>Subclass subgraphs automatically include the specified attributes of superclass subgraphs.

also, I think it's desirable to return an existing instance:

```suggestion
		final SubGraphImplementor<?> existingSubgraph = subclassSubgraphs.get( subTypeEntityDomainType );
		if ( existingSubgraph != null ) {
			return existingSubgraph;
		}
		final SubGraphImplementor<S> subgraph = new SubGraphImpl<>( subTypeEntityDomainType, true );
		subclassSubgraphs.put( subTypeEntityDomainType, subgraph );
		return subgraph;
```

Reply:
Done. 
I've started to link the subgraph to its parent. Does this mean that I have to override the findAttributeNode methods in AbstractGraph just to combine the attributes? Or should I also override attributeMap via a getter ?

Reply:
Hello @beikov ,
I have the impression that the link with the “superclass” is already somewhat taken into account by the “findAttributeNode” method. Even so, one solution I can see to better comply with the specification would be to use the constructor that takes in the “superclass” to hydrate attrNodeMap. What do you think ?